operator/pkg/controller/istiocontrolplane/istiocontrolplane_controller_test.go

```diff
@@ -65,6 +67,11 @@ type testCase struct {
 
 // TestIOPController_SwitchProfile
 func TestIOPController_SwitchProfile(t *testing.T) {
+	// Since controller code is running locally, we can point to a local filesystem path.
+	installPackagePath = filepath.Join(repoRootDir, "cmd/mesh/testdata/manifest-generate/data-snapshot")
```

Comment:
why is this using snapshot?

Reply:
because these tests are checking logic, not the templates.

Reply:
If its testing logic then why not use the real templates? If a change to the template breaks the test its likely a bug, right? I may not fully understand the test though. It seems the ones we want to use fake templates are ones that are literally just testing the templates and creating golden files. All others can use the real ones so that we identify issues immediately

Reply:
this is testing that the code installs/removes components correctly against some given set of templates. the form of the templates themselves is not very important here.
actually this test has limited value imo and we can probably get rid of it altogether when @richardwxn completes his e2e test. it was an early attempt at an e2e test but it doesn't do that. it's basically only guarding against some regressions at this point.